org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/view/task/TaskViewContentProvider.java

```diff
@@ -131,11 +142,34 @@ public boolean hasChildren(Object element) {
     }
 
     private Object[] childrenOf(ProjectNode projectNode) {
+        boolean isGroupProjects = this.taskView.getState().isGroupProjects();
+
+        return projectNodesFor(projectNode, isGroupProjects).toArray();
+    }
+
+    private Set<Object> projectNodesFor(ProjectNode projectNode, boolean isGroupProjects) {
+        Set<Object> result = Sets.newHashSet();
+
+        if (isGroupProjects) {
+            OmniEclipseProject eclipseProject = projectNode.getEclipseProject();
+            OmniGradleProject gradleProject;
+            Optional<IProject> workspaceProject;
```

Comment:
Why not initialize the variables inside the `for` loop? Anyway, they might disappear in the refactoring.

Reply:
I had a consideration in performance perspective. Actually, it's not good for readable.